src/lib/stores/wallet/wallet.store.ts

```diff
@@ -1,39 +1,42 @@
-import { ethers } from 'ethers';
+import { ethers, providers } from 'ethers';
 import type { Network } from '@ethersproject/networks';
-import Web3Modal from 'web3modal';
-import { writable } from 'svelte/store';
+import { get, writable } from 'svelte/store';
 import { browser } from '$app/environment';
-import { AddressDriverClient, Utils } from 'radicle-drips';
+import { AddressDriverClient } from 'radicle-drips';
 
 // https://github.com/vitejs/vite/issues/7257#issuecomment-1079579892
-import WalletConnectProvider from '@walletconnect/web3-provider/dist/umd/index.min.js';
 import testnetMockProvider from './__test__/local-testnet-mock-provider';
 import isTest from '$lib/utils/is-test';
-import { getAddressDriverClient } from '$lib/utils/get-drips-clients';
 import globalAdvisoryStore from '../global-advisory/global-advisory.store';
 
 import SafeAppsSDK from '$lib/stores/wallet/safe/sdk';
 import { SafeAppProvider } from '@safe-global/safe-apps-provider';
 import isRunningInSafe from '$lib/utils/is-running-in-safe';
 
+import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum';
+import { Web3Modal } from '@web3modal/html';
+import {
+  configureChains,
+  createConfig,
+  watchAccount,
+  watchWalletClient,
+  watchNetwork,
+  getNetwork,
+  type WalletClient,
+  getWalletClient,
+} from '@wagmi/core';
+import { mainnet, goerli } from '@wagmi/core/chains';
+import unreachable from '$lib/utils/unreachable';
+
 const appsSdk = new SafeAppsSDK();
 
-const { SUPPORTED_CHAINS } = Utils.Network;
 const DEFAULT_NETWORK: Network = {
   chainId: 1,
   name: 'homestead',
 };
 
-const INFURA_ID = 'aadcb5b20a6e4cc09edfdd664ed6334c';
-
-const WEB_3_MODAL_PROVIDER_OPTIONS = {
-  walletconnect: {
-    package: WalletConnectProvider,
-    options: {
-      infuraId: INFURA_ID,
-    },
-  },
-};
+const WAGMI_CHAINS = [mainnet, goerli];
+const WALLETCONNECT_PROJECT_ID = 'c09f5d8545d67c604ccf454219fd8f4d';
```

Comment:
Should we make this "private"?

Reply:
It's not secret, so I think we can keep it in here.

Reply:
(Didn't notice before approving but svelte-check's failing)

Reply:
yep i saw. should be resolved now